src/Microsoft.NetCore.Analyzers/Core/Security/DoNotDisableSchUseStrongCrypto.cs

```diff
@@ -0,0 +1,99 @@
+            SystemSecurityCryptographyResources.ResourceManager,
+            typeof(SystemSecurityCryptographyResources));
+        private static readonly LocalizableString s_Description = new LocalizableResourceString(
+            nameof(SystemSecurityCryptographyResources.DoNotDisableSchUseStrongCryptoDescription),
+            SystemSecurityCryptographyResources.ResourceManager,
+            typeof(SystemSecurityCryptographyResources));
+
+        internal static DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+                DiagnosticId,
+                s_Title,
+                s_Message,
+                DiagnosticCategory.Security,
+                DiagnosticSeverity.Error,
+                isEnabledByDefault: DiagnosticHelpers.EnabledByDefaultIfNotBuildingVSIX,
+                description: s_Description,
+                helpLinkUri: null,
+                customTags: WellKnownDiagnosticTags.Telemetry);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+
+            // Security analyzer - analyze and report diagnostics on generated code.
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
+
+            context.RegisterCompilationStartAction(compilationStartAnalysisContext =>
+            {
+                var compilation = compilationStartAnalysisContext.Compilation;
+                var appContextTypeSymbol = compilation.GetTypeByMetadataName(WellKnownTypes.SystemAppContext);
+
+                if (appContextTypeSymbol == null)
+                {
+                    return;
+                }
+
+                var setSwitchMemberWithStringAndStringParameter = appContextTypeSymbol.GetMembers("SetSwitch").OfType<IMethodSymbol>().GetSingleOrDefaultMemberWithParameterInfos(
+                                                                        GetParameterInfo(compilation.GetSpecialType(SpecialType.System_String)),
+                                                                        GetParameterInfo(compilation.GetSpecialType(SpecialType.System_Boolean)));
+
+                if (setSwitchMemberWithStringAndStringParameter == null)
+                {
+                    return;
+                }
+
+                compilationStartAnalysisContext.RegisterOperationAction(operationAnalysisContext =>
+                {
+                    var invocationOperation = (IInvocationOperation)operationAnalysisContext.Operation;
+                    var methodSymbol = invocationOperation.TargetMethod;
+
+                    if (methodSymbol.Equals(setSwitchMemberWithStringAndStringParameter))
+                    {
+                        var values = invocationOperation.Arguments.Select(s => s.Value.ConstantValue).ToArray();
```

Comment:
Nit: Could avoid a temporary array by just looking at invocationOperation.Arguments[0].ConstantValue, etc. directly.

Reply:
(Not really a big deal since it'll only be allocated on AppContext.SetSwitch() invocations.)